### PR TITLE
chore(ordered-values): added support for columns and column-rule

### PR DESCRIPTION
**Pull request by anikethsaha**
Ref #756

- `columns`
  In this, they takes two values, `width` and `count`, they both can be numbers and string(auto, inherit, unset etc. ) but `count` can't have `units`, and as per MDN's formal format, `width` should come first. So based on this, the function will change the order only if `unit` are mentioned.

- `column-rule`
  should be ordered like `border`

### CODE REVIEW

File: packages/postcss-ordered-values/src/rules/columns.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { unit } from 'postcss-value-parser';
+
+const strValues = ['auto', 'inherit', 'unset', 'initial'];
+
+export default (columns) => {
+  let newValue = { front: '', back: '' };
+  let shouldNormalize = false;
+  columns.walk((node) => {
+    const { type, value } = node;
+    if (type === 'word' && strValues.indexOf(value.toLowerCase())) {
```
Comment (jamesgeorge007) [inline packages/postcss-ordered-values/src/rules/columns.js]:
```suggestion
    if (type === 'word' && strValues.includes(value.toLowerCase())) {
```

Comment (anikethsaha) [reply packages/postcss-ordered-values/src/rules/columns.js]: I need to revert it, its failing the tests

Comment (jamesgeorge007) [reply packages/postcss-ordered-values/src/rules/columns.js]: Oops! I did found couple of instances where `Array.includes()` was used for the purpose.

Comment (anikethsaha) [reply packages/postcss-ordered-values/src/rules/columns.js]: 👍

Comment (jamesgeorge007) [reply packages/postcss-ordered-values/src/rules/columns.js]: https://github.com/cssnano/cssnano/search?q=includes&unscoped_q=includes

Comment (anikethsaha) [reply packages/postcss-ordered-values/src/rules/columns.js]: Not sure why then

File: packages/postcss-ordered-values/src/index.js [hunk 2]
```diff
@@ -22,13 +23,19 @@ const borderRules = {
   'border-left': border,
 };
 
+const columnRules = {
+  'column-rule': border,
```
Comment (evilebottnawi) [inline packages/postcss-ordered-values/src/index.js]: better create separate function, or rename `border` to `borderAndColumn`

File: packages/postcss-ordered-values/src/rules/columns.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    } else if (type === 'word') {
+      newValue.back = `${newValue.back} ${value}`;
+    }
+    // console.log({ type, value });
+  });
+  if (shouldNormalize) {
+    return `${newValue.front.trimStart()} ${newValue.back.trimStart()}`;
+  }
+  return columns;
+};
```
Comment (evilebottnawi) [inline packages/postcss-ordered-values/src/rules/columns.js]: Remove `// console.log`

File: packages/postcss-ordered-values/src/index.js (hunk 2 above)
Comment (anikethsaha) [reply packages/postcss-ordered-values/src/index.js]: cool 👍

File: packages/postcss-ordered-values/src/rules/columns.js (hunk 3 above)
Comment (anikethsaha) [reply packages/postcss-ordered-values/src/rules/columns.js]: sure 👍

**Comment by anikethsaha** [comment]
cc @evilebottnawi

File: packages/postcss-ordered-values/src/__tests__/index.js [hunk 4]
```diff
@@ -581,3 +581,19 @@ test(
     'border-inline: 5px dashed blue; border-inline-end: .5em solid red; border-block-end: 5px dashed blue; border-block: 5px solid red;border-block-start: 1px solid rgba(0, 30, 105, 0.8);'
   )
 );
+
+test(
+  'should order columns only if unit is mentioned',
+  processCSS(
+    'h1 {columns: 2 auto;columns: auto 12em;columns: auto auto;columns: auto 12em auto 13em;}',
```
Comment (ben-eb) [inline packages/postcss-ordered-values/src/__tests__/index.js]: Is `auto 12em auto 13em` a valid value? I'd be wary of adding tests around invalid syntax.

Comment (anikethsaha) [reply packages/postcss-ordered-values/src/__tests__/index.js]: no it's not, sorry !!!
I will change it

**Comment by anikethsaha** [comment]
cc @ben-eb @evilebottnawi

**Comment by evilebottnawi** [comment]
Thanks!

**Pull request merged**